server/src/test/java/org/elasticsearch/index/engine/LiveVersionMapTests.java

```diff
@@ -137,24 +142,25 @@ public void testConcurrently() throws IOException, InterruptedException {
                             }
                             if (isDelete == false && rarely()) {
                                 versionValue = new DeleteVersionValue(versionValue.version + 1, versionValue.seqNo + 1,
-                                    versionValue.term, Long.MAX_VALUE);
+                                    versionValue.term, clock.incrementAndGet());
                             } else {
                                 versionValue = new VersionValue(versionValue.version + 1, versionValue.seqNo + 1, versionValue.term);
                             }
                             values.put(bytesRef, versionValue);
                             map.putUnderLock(bytesRef, versionValue);
                         }
+                        if (rarely()) {
+                            map.pruneTombstones(0, 0);
+                        }
                     }
                 } finally {
                     done.countDown();
                 }
             });
             threads[j].start();
-
-
         }
         do {
-            Map<BytesRef, VersionValue> valueMap = new HashMap<>(map.getAllCurrent());
+            final Map<BytesRef, VersionValue> valueMap = new HashMap<>(map.getAllCurrent());
```

Comment:
I think this is now less strong as we don't track deletes here any more... we should somehow show that we capture deletes correctly in the various phases of refresh and that we don't forget them until we prune. Please me correct if I'm missing something. This is tricky ;)